### PR TITLE
Fix regex strings in gitpod-cli and ws-daemon

### PR DESCRIPTION
**Pull request by saisrivathsa**

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4843#" title="Author self-approved">saisrivathsa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **fntlnz** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @fntlnz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-cli/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-cli/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by saisrivathsa** [comment]
Created a new PR with requested changes [here](https://github.com/gitpod-io/gitpod/pull/3679)

File: components/ws-daemon/pkg/quota/size.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ const (
 )
 
 var (
-	sizeRegexp = regexp.MustCompile(`(\d+)(k|m|g|t)?`)
+	sizeRegexp = regexp.MustCompile(`^(\d+)(k|m|g|t)?$`)
```
Comment (saisrivathsa) [inline components/ws-daemon/pkg/quota/size.go:34]: Made changes as requested by @corneliusludmann  [here](https://github.com/gitpod-io/gitpod/pull/3679#discussion_r629290934)

File: components/ws-daemon/pkg/quota/size_test.go [hunk 2]
```diff
@@ -23,8 +23,8 @@ func TestParseSize(t *testing.T) {
 		{"42m", 42 * quota.Megabyte, nil},
 		{"42g", 42 * quota.Gigabyte, nil},
 		{"42t", 42 * quota.Terabyte, nil},
-		// This test should fail but doesn't because match somehow matches the regexp
-		// {"-42m", 0, quota.ErrInvalidSize},
+		{"-42m", 0, quota.ErrInvalidSize},
```
Comment (saisrivathsa) [inline components/ws-daemon/pkg/quota/size_test.go:26]: Made changes as requested by @corneliusludmann  [here](https://github.com/gitpod-io/gitpod/pull/3679#discussion_r629290934)

**Comment by csweichel** [comment]
@aledbf do you know `shellcheck` would fail here? There's no change to this file in this PR.

**Comment by aledbf** [comment]
@csweichel fix in https://github.com/gitpod-io/gitpod/pull/4850

**Comment by aledbf** [comment]
@saisrivathsa please rebase

**Pull request closed without merging**